tapestry.js

```diff
@@ -975,41 +1005,102 @@ function getBoundedCoord(coord, maxCoord) {
     return Math.max(MAX_RADIUS, Math.min(maxCoord - MAX_RADIUS, coord));
 }
 
-function getChildren(id) {
-    var children = [];
-    var dataLinks = dataset.links;
-    for (var linkId in dataLinks) {
-        var link = dataLinks[linkId];
-        //SEARCH FOR: CURRENT_ID_NODE -> CHILD links
-        if (typeof link.source === 'number' && link.source === id) {
-            children.push(link.target);
-        } else if (typeof link.source === 'object' && link.source.id === id) {
-            children.push(link.target.id);
-        }
+// add 'depth' parameter recursively to each node
+function addDepth(rootId, depth, visitlist) {
+    var visited = visitlist;
```

Comment:
Instead of creating an additional variable, just change your 3rd parameter to `visited`.